File: max_sum_without_adjacent.py
# User function Template for python3
class Solution:

    def findMaxSum(self, arr, n):
        # code here
        curr, prev = arr[0], 0
        for i in range(1, n):
            if prev+arr[i] > curr:
                prev, curr = curr, prev+arr[i]
            else:
                prev = curr

        return max(curr, prev)


# {
 # Driver Code Starts
# Initial Template for Python 3

if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n = int(input())
        arr = list(map(int, input().strip().split()))
        ob = Solution()
        ans = ob.findMaxSum(arr, n)
        print(ans)
        tc -= 1

# } Driver Code Ends

File: max_sum_without_adjacent.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function template for C++
class Solution
{
public:
    // calculate the maximum sum with out adjacent
    int findMaxSum(int *arr, int n)
    {
        // code here
        if (n == 1)
            return arr[0];
        int prev = 0, curr = arr[0];
        for (int i = 1; i < n; i++)
        {
            if (prev + arr[i] > curr)
            {
                int t = curr;
                curr = prev + arr[i];
                prev = t;
            }
            else
            {
                prev = curr;
            }
        }
        return max(prev, curr);
    }
};

//{ Driver Code Starts.

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++)
        {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.findMaxSum(arr, n);
        cout << ans << "\n";
    }
    return 0;
}
// } Driver Code EndsF